crates/forge/src/main.rs

```diff
@@ -166,6 +167,17 @@ fn test_workspace(args: TestArgs) -> Result<bool> {
         bail!("Scarb build did not succeed")
     }
 
+    let test_build_output = Command::new("scarb")
+        .arg("snforge-test-collector")
+        .env("SCARB_PACKAGES_FILTER", filter_env)
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .output()
+        .context("Failed to build test artifacts with Scarb")?;
+    if !test_build_output.status.success() {
+        bail!("scarb snforge-test-collector did not succeed")
+    }
```

Comment:
this deserves own function

Reply:
Then we want `scarb build` in its own function too?